portfolio/src/main/webapp/script.js

```diff
@@ -60,6 +51,62 @@ async function getCommentFromFetch() {
         var para = document.createElement("p");
         var node = document.createTextNode(commentString);
         para.appendChild(node);
+        var button = document.createElement("button");
+        button.innerHTML = "Like";
+        button.type = "button";
+        const input = commentString.slice();
+        button.addEventListener("click", function () { likeComments(input) });
         commentContainer.appendChild(para);
+        commentContainer.appendChild(button);
     };
+}
+
+async function fetchCommentsWithStoredLimit() {
+    console.log("Try fetch");
+    const response = await fetch("/data");
+    const commentResponse = await response.json();
+    const comments = commentResponse.commentList;
+    var maxComments = parseInt(commentResponse.maxComments, 10);
+    if (Number.isNaN(maxComments)) {
+        console.log("You are trying to input a non-numeric data, please try again");
+        // Set the limit to default value
+        maxComments = 10;
+    }
+    const commentLimitContainer = document.getElementById("comment-limit");
+    commentLimitContainer.value = (maxComments.toString());
+    displayComments(maxComments, comments);
+}
+
+async function fetchCommentsWithInputLimit() {
+    console.log("Try fetch");
+    const response = await fetch("/data");
+    const commentResponse = await response.json();
+    const comments = commentResponse.commentList;
+    var maxComments = parseInt(document.getElementById("comment-limit").value, 10);
+    if (Number.isNaN(maxComments)) {
+        console.log("You are trying to input a non-numeric data, please try again");
+        // Set the limit to default value
+        maxComments = 10;
+    }
+    if (maxComments < 1) {
+        console.log("Please input a positive integer");
+        // Set the limit to default value
+        maxComments = 10;
+    }
+    displayComments(maxComments, comments);
+}
```

Comment:
These two functions would benefit from being reduced into a single function, since they largely do the same thing. The second one has an extra bounds-check on `maxComments` because it's user provided, but there's no reason not to do the same for the `maxComments` provided by the server. Imagine the server getting updated someday. If there's a bug in the backend method that causes it to return out-of-bounds `maxComments` values to your JS, you'd be protected from that.

The other difference is that you update the value in the input element; one simple way to reconcile the two functions would be to first check if the current value of the input doesn't match what you've decided the `maxComments` value is; only update the value in the input if they disagree.

I believe those two changes will allow you to use the same JS function for both features.

Reply:
I think it over and still think they should be separated because the situation that the two functions are called is different. When `fetchCommentsWithStoredLimit` is called, the page has just been reloaded and the stored `maxComments` value needs to be kept because I want to restore the value of  `maxComments`. And under such situation, the value in the input element is actually the default value. So when the two values are different, we want to keep the stored value.

However, when `fetchCommentsWithInputLimit` is called, the user just input a new value and we need to update the value in the input if the input value and stored `maxComments` disagree. Under such situation, we will want to keep the newly input value. 

Thus, the function needs to behave in a different way when the two values disagree and it makes it hard to combine the two functions.

Reply:
Ah, yes, you're right. I was still thinking of the prior version of the site where the max comments input was submitted along with the form to add a new comment.

Consider this, though: what if `fetchCommentsWithInputLimit` told the server what the new user-provided limit was?

Reply:
I think if I told the server what the new user-provided limit was, I can largely reuse the `fetchCommentsWithStoredLimit` function. Although I may still need two separate functions, but the `fetchCommentsWithInputLimit` will just tell the server what the new user-provided limit is and call the `fetchCommentsWithStoredLimit`.

Reply:
In the latest commit, I managed to combine the two functions as much as possible. For now I still haven't found a method that can truly reduce the two functions into one single function but I try to increase the code reuse as much as possible.